source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.rst

```diff
@@ -612,6 +614,11 @@ The unstructured form SHOULD be as follows, using the values defined in the stru
   Heartbeat started for {{serverHost}}:{{serverPort}} on connection with driver-generated ID {{driverConnectionId}} and server-generated ID
   {{serverConnectionId}} in topology with ID {{topologyId}}. Awaited: {{awaited}}
 
+If the ``serverConnectionId`` is not yet known at the time this log is emitted, the unstructured form SHOULD be as follows:
+
+  Heartbeat started for {{serverHost}}:{{serverPort}} on connection with driver-generated ID {{driverConnectionId}} in topology with ID
+  {{topologyId}}. Awaited: {{awaited}}
```

Comment:
Is this documenting behavior that's already tested? Or do we need a new test for this?

Reply:
I'd note that this seems unrelated to DRIVERS-2677. Can we open a new drivers ticket for this change?

Reply:
This is new behaviour that needs to be tested, but can't be until a new DRIVERS ticket to add `serverConnectionId` to `ServerHeartbeatStartedEvent` as an optional field is created. I'm open to removing this until that ticket is filed and in progress.

Reply:
Cool, I vote we move it out to that ticket to avoid confusion.